catalog/app/components/Preview/quick/Markdown/Render.tsx

```diff
@@ -0,0 +1,28 @@
+import * as React from 'react'
+import * as Lab from '@material-ui/lab'
+
+import { useMarkdownRenderer } from 'components/Preview/loaders/Markdown'
+import Markdown from 'components/Preview/renderers/Markdown'
+import type * as Model from 'model'
+import AsyncResult from 'utils/AsyncResult'
+
+import Skeleton from './Skeleton'
+
+interface RenderProps {
+  value: string
+  handle: Model.S3.S3ObjectLocation
+}
+
+export default function Render({ value, handle }: RenderProps) {
+  const result = useMarkdownRenderer(AsyncResult.Ok(value), handle)
+  return AsyncResult.case({
+    _: () => null,
+    Err: (error: Error) => <Lab.Alert severity="error">{error.message}</Lab.Alert>,
+    Pending: () => <Skeleton />,
```

Comment:
it's probably better to just show skeleton on `_` (combining Pending and Init)

Reply:
Actually, `Pending` is redundant, because we always pass `AsyncResult.Ok`

But if there was a real request, it seems like a shortcut to combine `Init` and `Pending`.
Practically, `Init` is usually followed by `Pending` in requests. But what if someone put `noAutoFetch` somewhere in requests, and request will stack in `Init`? Probably, it would not be the desired UI state: showing skeleton when request is not pending.

Reply:
> it seems like a shortcut to combine Init and Pending.

well, exactly

> But what if someone put noAutoFetch somewhere in requests, and request will stack in Init? Probably, it would not be the desired UI state: showing skeleton when request is not pending.

well, that depends on the context. if you want to handle "init" state in a special way, e.g. show a distinct message or a special control -- that's ok. but if you know that there's no practical gain in handling those individually, then it's ok to combine them, and it's definitely better than just showing nothing (returning null).

ofc it's also ok to return null for any of those if you're absolutely sure they are not used, tho i'd prefer to use smth like `absurd` / `assertNever` in such cases to explicitly highlight the expected invariant